edx_management_commands/management_commands/management/commands/manage_user.py

```diff
@@ -0,0 +1,131 @@
+"""
+Management command `manage_user` is used to idempotently create or remove
+Django users, set/unset permission bits, and associate groups by name.
+"""
+
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Group
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from django.utils.translation import gettext as _
+
+
+@transaction.atomic
+class Command(BaseCommand):
+    # pylint: disable=missing-docstring
+
+    help = 'Creates the specified user, if it does not exist, and sets its groups.'
+
+    def add_arguments(self, parser):
+        parser.add_argument('username')
+        parser.add_argument('email')
+        parser.add_argument('--remove', dest='is_remove', action='store_true')
+        parser.add_argument('--superuser', dest='is_superuser', action='store_true')
+        parser.add_argument('--staff', dest='is_staff', action='store_true')
+        parser.add_argument('-g', '--groups', nargs='*', default=[])
+
+    def _maybe_update(self, user, attribute, new_value):
+        """
+        DRY helper.  If the specified attribute of the user differs from the
+        specified value, it will be updated.
+        """
+        old_value = getattr(user, attribute)
+        if new_value != old_value:
+            self.stderr.write(
+                _('Setting {attribute} for user "{username}" to "{new_value}"').format(
+                    attribute=attribute, username=user.username, new_value=new_value
+                )
+            )
+            setattr(user, attribute, new_value)
+
+    def _check_email_match(self, user, email):
+        """
+        DRY helper.
+
+        Requiring the user to specify both username and email will help catch
+        certain issues, for example if the expected username has already been
+        taken by someone else.
+        """
+        if user.email != email:
+            # The passed email address doesn't match this username's email address.
+            # Assume a problem and fail.
+            raise CommandError(
+                _(
+                    'Skipping user "{}" because the specified and existing email '
+                    'addresses do not match.'
+                ).format(user.username)
+            )
+
+    def _handle_remove(self, username, email):
+
+        users = get_user_model().objects.filter(username=username)
+        if not users:
+            self.stderr.write(_('Did not find a user with username "{}" - skipping.').format(username))
+            return
+
+        user = users[0]
```

Comment:
Just to avoid this, why not `users = get_user_model().objects.get(username=username)` and catch `DoesNotExist`?